packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol

```diff
@@ -0,0 +1,2926 @@
+                    .slotsBitmapB
+                    .minBitSet(
+                        // start at beginning of second bitmap
+                        uint8(0)
+                    );
+                // add 256 to account for slotsBitmapB offset
+                minSlotWithBid_ += 256;
+            }
+        }
+        // populate return value
+        if (!foundSlotWithBid) {
+            return uint16(NUM_SLOTS);
+        } else {
+            minSlotWithBid = uint16(minSlotWithBid_);
+            return minSlotWithBid;
+        }
+    }
+
+    /**
+     * @notice Helper function to get maximum slot index with an active bid,
+     * starting at a given slot index and searching downwards.
+     * Returns 512, (invalid slot index) if no slots with bids were found.
+     * Reverts if startSlotIndex > 511, since this library only supports 512
+     * slots.
+     * @param RAMProjectConfig_ RAM project config to query
+     * @param startSlotIndex Slot index to start search at
+     * @return maxSlotWithBid Maximum slot index with an active bid, and 512 (invalid index) if
+     * no slots with bids were found.
+     */
+    function _getMaxSlotWithBid(
+        RAMProjectConfig storage RAMProjectConfig_,
+        uint16 startSlotIndex
+    ) private view returns (uint16 maxSlotWithBid) {
+        bool foundSlotWithBid;
+        // revert if startSlotIndex > 511, since this is an invalid input
+        // @dev no coverage on if branch because unreachable as used
+        if (startSlotIndex > 511) {
+            revert("Only start slot index lt 512");
+        }
+        // temporary uint256 in working memory
+        uint256 maxSlotWithBid_;
+        // start at startSlotIndex
+        if (startSlotIndex < 256) {
+            // @dev <256 conditional ensures no overflow when casting to uint8
+            (maxSlotWithBid_, foundSlotWithBid) = RAMProjectConfig_
+                .slotsBitmapA
+                .maxBitSet(uint8(startSlotIndex));
+        } else {
+            // need to potentially check both bitmaps
+            (maxSlotWithBid_, foundSlotWithBid) = RAMProjectConfig_
+                .slotsBitmapB
+                .maxBitSet(
+                    // @dev casting to uint8 intentional overflow instead of
+                    // subtracting 256 from slotIndex
+                    uint8(startSlotIndex)
+                );
+            // add 256 to account for slotsBitmapB offset
+            maxSlotWithBid_ += 256;
+            if (!foundSlotWithBid) {
+                // no bids in first bitmap B, so check second bitmap A
+                (maxSlotWithBid_, foundSlotWithBid) = RAMProjectConfig_
+                    .slotsBitmapA
+                    .maxBitSet(
+                        // start at the end of the first bitmap
+                        uint8(255)
+                    );
+            }
+        }
+        // populate return value
+        if (!foundSlotWithBid) {
+            return uint16(NUM_SLOTS);
```

Comment:
missing if branch coverage here - should either add test for coverage or mark as unreachable as used

Reply:
Added a dev comment here 🙏